### ISSUE TITLE
Add epics and add issues to epics from Github

### ISSUE THREAD

**Issue by ac21** [codetree/feedback#304]
## Background
Some users prefer to interact with Codetree primarily through the GitHub issues interface.  However, you can not create epics or add issues to epics from GitHub at this time

## Solution
* Detect when a user adds the label `codetree-epic` to an issue and recognize that as an epic going forward
* detect when a comment or description entry of an epic includes the phrase `epic_child #(/d)` and add the referenced issue to the epic 
* detect when a comment or description entry of an issue includes the phrase `epic #(/d)` and add the  issue to the referenced epic

**Comment by arkan** [comment]
Any update regarding this?

**Comment by tomasyany** [comment]
Any update?

**Comment by ac21** [comment]
Hey @tomasyany - Thanks for reaching out.  This is still in our backlog.   Is this a high priority for your team?  would you say that most of your team works from within Github directly?

**Comment by tomasyany** [comment]
Hi @ac21! Most of my team works on Github directly indeed, when it comes to defining issues, adding tags, etc.

Nevertheless, I don't think that "adding Epics from Github" is _really_ the feature we need here, but rather "converting existing issues into Epics", something you can't do from Github nor Codetree. This feature could be really useful because it would allow not to lose all the discussion thread that already existed in the issue.

The solution proposed by the OP would kill two birds with one stone: adding epics from Github _and_ converting existing issues into Epics.

**Comment by Drjacky** [comment]
@tomasyany 
> The solution proposed by the OP would kill two birds with one stone: adding epics from Github and converting existing issues into Epics.

But how? 🤔

**Comment by tomasyany** [comment]
> @tomasyany
> 
> > The solution proposed by the OP would kill two birds with one stone: adding epics from Github and converting existing issues into Epics.
> 
> But how? 🤔

@Drjacky what part isn't clear? The "solution" (as in this issue description), seems to be a good answer to the "how" I believe...

**Comment by tomasyany** [comment]
Is there any update on this issue?

It would be really helpful if we could just add a label to an epic, or a comment on the issue, or some tag on the description, that would make that issue part of an epic directly.

**Comment by triclambert** [comment]
It looks like adding the `codetree-epic` label works to convert an issue to an epic from the GH UI.

**Comment by mikebessuille** [comment]
Yes, adding codetree-epic in GH seems to force Codetree to treat the issue as an epic.  However, removing the codetree-epic label does _not_ seem to convert the epic back to an issue!  Once it becomes an epic, I don't see any way of making it an issue again.
